### DISCUSSION TITLE
Feature: @mux/mux-video-react should provide playback engine handle

### DISCUSSION THREAD

**Discussion by shrugs** [muxinc/elements#399] [Ideas]
## Context

I'm trying to ensure that the user is always playing the latest of an `ll-live` stream, and to do so i'd like to seek to the hls live position as the playback core code does whenever the user pauses the stream, but this requires a handle to the playback engine, which the react bindings don't provide.

https://github.com/muxinc/elements/blob/main/packages/playback-core/src/autoplay.ts#L107

## Feature Request

the react bindings should include a handle to the playback engine so that i can write code like

```
useEffect(() => {
  if (!playing) ref.current.currentTime = ref.current.playbackEngine.liveSyncPosition;
}, [])
```

**Comment by luwes** [comment]
@shrugs thanks for the feedback! there is a undocumented `muxPlayer._hls` property but keep in mind that it could be changed in the future with a new player version.

moving this issue to discussion (idea) because we might want to expose a getter for `liveSyncPosition`.

**Comment by shrugs** [comment]
ah thanks i keep forgetting discussions are a thing. i'll definitely use that _hls handle then! it could be neat to have a "live" prop on the element that ensures this behavior

**Reply by cjpillsbury** [reply]
I like this idea... you can also get _**pretty close**_ by using `muxPlayer.media.seekable.end(0)` for your reference time. I'm also adding a ticket to make that part of the public/top level API. 😎

**Reply by gkatsev** [reply]
seekable end has the benefit of working with native playback too!

**Comment by shrugs** [comment]
I ended up switching to a "always playing (initially muted), never pausable" mechanic with the new react player and don't explicitly need this feature anymore, but could be a nice library improvement
